### PR TITLE
Event Logger

### PR DESCRIPTION
**Pull request by kinrezC**
**Give an overview of the tasks completed**
Introduces `EventLogger`.  `EventLogger` has an `EventLoggerBuilder` which allows one to chain `EventLogger::builder().add(contract0.events()).add(contract1.events())` to listen for events on specified contracts.  When the `EventLogger` is `run().await` it will watch for events and write them to a csv.  

**Link to issue(s) that this PR closes**
For #515

### CODE REVIEW

**Review by 0xJepsen — APPROVED** [review]
This is so clean @kinrezC! Awesome work on this! You are officially the async rust chad.

File: arbiter-core/src/tests/data_output.rs
Comment (Autoparallel) [inline arbiter-core/src/tests/data_output.rs:1]: My comments are mostly here.

Can we perhaps have these tests check that the files are actually created and have the correct data in them?

We don't need to take it super far, but I feel like these don't check all that much other than that this runs.

**Review by Autoparallel — CHANGES_REQUESTED** [review]
How is the stop signal sent to the data collector? Do we need that ever? Can you explain why/why not?

**Review by Alexangelj — APPROVED** [review]
Looks excellent

**Pull request merged**